### PR TITLE
Fix for wrong usage of memset and few unbalanced python refcount

### PR DESCRIPTION
**Pull request by hirenvadalia**
#### Bug/feature Description
* *Wrong usage of memset before check for null after memory allocation*
* *unbalanced python object refcount*

#### Affected Platform(s)
* *All*

#### Cause / Analysis / Design
* *None*

#### Solution Description
* *Corrected usage of memset aka moved memset after null check after memory allocation*
* *Corrected unbalanced python object reference count*

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
   - [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
   - [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [x] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

**Comment by hirenvadalia** [comment]
Since this is fix for wrong usage of memset I believe we don't need any new test, please let me know if anyone think otherwise.

File: src/lib/Libpython/pbs_python_svr_internal.c [hunk 1]
```diff
@@ -836,9 +837,8 @@ pbs_python_setup_resv_class_attributes(void)
 	else
 		DEBUG2_ARG1("BEGIN setting up all reservation attributes %s", "");
 	py_resv_attr_types = PyMem_New(PyObject *, num_entry);
-	memset(py_resv_attr_types, 0, sizeof(PyObject *)*num_entry);
-
 	if (!py_resv_attr_types) { goto ERROR_EXIT;}
+	memset(py_resv_attr_types, 0, sizeof(PyObject *)*num_entry);
```
Comment (bhroam) [inline src/lib/Libpython/pbs_python_svr_internal.c]: Please put spaces around the '*' between the sizeof*() and num_entry (same for the rest of the memset() calls)

Comment (mike0042) [reply src/lib/Libpython/pbs_python_svr_internal.c]: Good point. They look like dereferences without the spaces.

Comment (hirenvadalia) [reply src/lib/Libpython/pbs_python_svr_internal.c]: @bhroam and @mike0042 Thanks fore review. I have addressed your comment. Please have a look into latest commit.

**Comment by mike0042** [comment]
Looks good.

**Comment by bhroam** [comment]
Thanks for making the change, @hirenvadalia.  Looks good to me too.

**Comment by hirenvadalia** [comment]
Thanks @arungrover, @vstumpf, @mike0042 and @bhroam for review. I have squashed and updated with latest master, now is ready for merge.

**Pull request merged**